# HG changeset patch
# User alta88@fixall.com
# Date 1740625508 25200
# Parent  a01054003da5f9b27a9b42fc21e9502efea52f12
Feed enhancements
- Implement UIDensity in Subscribe dialog.

diff --git a/mail/themes/shared/mail/feedSubscribe.css b/mail/themes/shared/mail/feedSubscribe.css
--- a/mail/themes/shared/mail/feedSubscribe.css
+++ b/mail/themes/shared/mail/feedSubscribe.css
@@ -167,4 +167,29 @@ label:has([hidden]) {
 
 #autotagPrefix {
   max-width: 15em;
 }
+
+/* UI Density customization - match folderPane. */
+
+:root[uidensity="compact"] {
+  & #folderTree > #folderTreeChildren::-moz-tree-row {
+    min-height: 20px;
+  }
+  & #feedDetailGrid {
+    grid-gap: 0;
+  }
+  & #feedOptionsGrid {
+    grid-gap: 4px;
+  }
+}
+:root[uidensity="touch"] {
+  & #folderTree > #folderTreeChildren::-moz-tree-row {
+    min-height: 33px;
+  }
+  & #feedDetailGrid {
+    grid-gap: 2px;
+  }
+  & #feedOptionsGrid {
+    grid-gap: 10px;
+  }
+}
diff --git a/mailnews/extensions/newsblog/feed-subscriptions.js b/mailnews/extensions/newsblog/feed-subscriptions.js
--- a/mailnews/extensions/newsblog/feed-subscriptions.js
+++ b/mailnews/extensions/newsblog/feed-subscriptions.js
@@ -26,8 +26,11 @@ var { FileUtils } = ChromeUtils.importES
 );
 var { PluralForm } = ChromeUtils.importESModule(
   "resource:///modules/PluralForm.sys.mjs"
 );
+var { UIDensity } = ChromeUtils.importESModule(
+  "resource:///modules/UIDensity.sys.mjs"
+);
 var { UIFontSize } = ChromeUtils.importESModule(
   "resource:///modules/UIFontSize.sys.mjs"
 );
 var { XULStoreUtils } = ChromeUtils.importESModule(
@@ -112,8 +115,9 @@ var FeedSubscriptions = {
       "accountLabel",
       FeedUtils.stringsMessenger.GetStringFromName("removeAccount")
     );
 
+    UIDensity.registerWindow(window);
     UIFontSize.registerWindow(window);
   },
 
   onDialogAccept() {
